Biodiversity Genomics 2020

**Digging into the capybara and lesser capybara genomes - Detailed methods**

María José Gómez-Hughes, Santiago Herrera-Álvarez & Andrew J. Crawford

**_Genome & Transcriptome assemblies_**
For the capybara (H. hydrochaeris), we used the genome assembly and annotation by Herrera-Álvarez et al. (2020). In brief, tissue from a wild-caught but captive-raised capybara, reportedly from Paraguay, was donated by the San Diego Zoo’s Frozen Zoo to the 200 Mammalian Genomes Project led by the Broad Institute, which then sequenced and assembled a draft genome using the DISCOVAR de novo sequencing and assembly method (Weisenfeld et al., 2014). This assembly was then ‘upgraded’ using Chicago libraries provided by Dovetail Genomics (Putnam et al., 2016). 
For the lesser capybara genome assembly, tissue samples were collected from one juvenile H. isthmius from San Juan del Carare, Santander, Colombia (Lat: 6.72, Lon: -74.15), on June 22nd,  2017, that was subsequently accessioned into the Mammal collection of the Museo Historia Natural C.J. Marinkelle, Universidad de los Andes, Bogotá, Colombia (field number AJC 7100, voucher number ANDES-M 2300). This sample was sequenced using 10X Genomics linked reads technology in two lanes of Illumina HiSeq X10.
The resulting reads from the sequencing were run through Long Ranger v2.2.2 (10X Genomics) to process barcodes. These reads were assembled with Supernova v2.0.1 (Weisenfeld et al., 2017), an assembler created by 10X Genomics that uses a progressively larger contigs approach and its own trimming step to create phased scaffolds from the reads. We included the mkoutput pseudohap option to visualize only one hypothetical haplotype on the resulting assembly. 
To enhance this assembly we used the following four steps. 1) Tigmint v1.1.2 was used to  produce an assembly that is both more contiguous and more correct by comparing the alignment of linked reads to the draft assembly to correct for possible mis-assemblies (Jackman et al., 2018). 2) Arcs v1.0.6 was used to scaffold the assembly (Yeo et al., 2017). 3) Sealer v2.0.2 was used to identify intra-scaffold gaps in the draft assembly, search for flanking sequences, and then to fill the gaps by realigning the raw reads to the assembly (Paulino et al., 2015). Sealer navigates de Bruijn graphs via bloom filters based on k values and we chose k values of 64, 80, 96, 112, and 128, since this would give us a range that could help us close gaps on areas of low coverage with the lower values, and areas of high repetition levels with the higher values (Paulino et al., 2015). 4) Purge_dups  v1.2.3 was used to remove duplicates due to assembling errors by searching for similarity between contigs and read depths. To remove duplicates, we chose a low read-coverage of 0.5, a middle coverage of 25 and a high coverage of 220, contigs below the low coverage and over the high coverage were discarded by being considered “junk” and duplicated contigs respectively (Guan et al., 2020). 
Transcriptomic data were obtained from eleven tissues representing one male and one female H. hydrochaeris from eastern Colombia. Tissues were preserved in Nucleic Acid Preservation buffer (NAP; Camacho-Sánchez et al., 2013) to avoid RNA degradation. RNA was extracted with standard TRI Reagent® Solution (Ambion Inc., Austin, Texas, USA), cleaned using the RNeasy Plus Mini Kit (Qiagen. Hilden, Germany), and diluted to a final volume of 30μl with nuclease-free water. Quantity of extracted RNA for library construction was measured with the Qubit® RNA HS Assay kit. Complementary DNA (cDNA) libraries were constructed with the Illumina TruSeq v2 kit using half reactions. Quality of cDNA libraries was assessed using Agilient High Sensitivity DNA kits on a 2100 BioAnalyzer. The 11 libraries were barcoded and run together in paired-end mode on one lane of an Illumina Hiseq 2000.
For the transcriptome assembly, we used all reads from the 11 sequenced libraries. We first trimmed the reads with trimmomatic v0.39 (Bolger, Lohse & Usadel, 2014), filtered with the FASTX-toolkit v0.0.14, normalized based on the median coverage, and trimmed unreliable k-mers using the khmer v1 digital normalization algorithm (Crusoe et al., 2015; Brown et al., 2012). We then used Trinity to assemble the transcriptome from the remaining reads with default parameters (Grabherr et al., 2011).  
To assess genome completeness on all the assemblies, we ran BUSCO v3.0.2 using the Vertebrate dataset (Waterhouse et al., 2017).

**_Genome & Transcriptome annotations_**
To repeat mask the H. isthmius assembly, we took two approaches: a de novo a approach and a reference-based approach. For the first approach, we used RepeatModeler v2.0.1 to model families of transposable elements (TE; Flynn et al., 2020). We then fed the resulting library of TEs to RepeatMasker v4.0.9_p2 (Smit, Hubley & Green, 2015). For the second approach, we again used RepeatMasker, but specifying “rodentia” as the species to guide the masking using repeat evidence from other rodents. We then concatenated both outputs and used them to mask the primary assembly using the program ProcessRepeats from RepeatMasker. For the type of masking, we chose a soft-masked approach which gave us the possibility to visualize what the repeat subsequences were, but without them interfering in downstream analyses.
For annotating the H. isthmius  genome, we selected three high-quality, annotated genome assemblies from representative rodent species. We used the Maker v2.31 pipeline (Holt & Yandell, 2011) based on proteomes from the guinea pig (Cavia porcellus, Cavpor 3.0), the house mouse (Mus musculus; GRCm38.p6), and the common rat (Rattus novergicus; Rnor6.0) to guide the annotation. We downloaded the proteomes from the Ensembl release 97 and clustered them with CD-HIT v4.6.1 into a single non-redundant file (Li, Jaroszewski, & Godzik, 2001). We used the  Swiss-Prot reviewed database (UniProt Consortium, 2018) to add functionality to the annotations of both species by homology hits found by Blastp v2.9.0 with a evalue of 10⁶. Additionally, we used InterProScan v5.36-75 (Jones et al., 2014) to classify the annotated genes into Pfam protein families (El-Gebali et al, 2018).
To functionally annotate the transcriptome, we extracted the longest open reading frames and predicted the most likely coding regions with Transdecoder v3.0.0 (Haas & Papanicolaou, 2015).  Then we used Trinotate to functionally annotate the predicted polypeptides and to create a database for navigating these data (following Bryant et al., 2017). We used BLAST v2.9.0 to search for homology hits against the UniProt Swiss-Prot database (UniProt Consortium, 2018), and identify to which Pfam protein family each transcript belonged (El-Gebali et al., 2018) using profile hidden Markov models with HMMER v3.2.1 (Mistry et al., 2013) , and assigned inferred proteins to Eggnog functional categories (Huerta-Cepas et al., 2015) and to gene ontology categories (GO; Ashburner et al., 2000) using BLAST v2.9.0.

**_Demography, divergence & diversity_**
As an effort to aid in conservation efforts and the urgency of these, we looked at changes in effective population size from the past towards the present. We used pairwise sequentially Markovian coalescent (PSMC; Li & Durbin, 2011) to infer how the effective population size (Ne ) may have changed over recent history in the two species of Hydrochoerus. Briefly, this algorithm reconstructs the distribution of times to the most recent common ancestor (TMRCA) along chromosomes by examining the density of heterozygous sites (Li & Durbin, 2011). To do this, we first indexed the H. hydrochaeris assembly and mapped the raw reads of both species independently back to their own assembly to avoid technical errors and mismatches due to divergence with bwa v0.7.4 (Li & Durbin, 2009). We then sorted each alignment and converted it from a BAM to a VCF file with SAMtools v1.8 (Li et al., 2009), called the SNPs and indels with bcftools v1.8 (Li, 2011), and transformed this file to a fastq file with vcftools v4.2.0 (Danecek et al, 2011). We then used this file to estimate the parameters of the PSMC model with 25 iterations, and assuming a generation time of two and three years, a mutation rate of 2.2 ×10-9, and a recombination parameter of “4+25*2+4+6” implemented in psmc v0.6.5 (Li & Durbin, 2011). The mutation rate used is assumed to be the average mutation rate of all mammals as stated by Kumar & Subramanian (2002). 
To estimate divergence time within Hydrochoerus species, we put them in a phylogenetic context with other Hystricomorpha species, we downloaded from Ensembl all available proteomes of rodent species, and from the european rabbit, Oryctolagus cuniculus, for a total of 27 species and used Orthofinder v2.3.3 to search for orthologs (Emms, & Kelly, 2019). Then, to put them in a phylogenetic context, we extracted all single copy orthologs that were found in all nine species and performed a pre-alignment quality filter with PREQUAL v1.02 to identify and filter non-homologous sequences (Whelan, Irisarri & Burki, 2018). We performed a multiple sequence alignment with MAFFT v7.309 assuming a BLOSUM62 matrix in each orthologue independently (Katoh & Standley, 2013), and then trimmed the alignments for poorly aligned regions with trimAl v1.4 to maintain only the most reliable alignments (Capella-Gutiérrez, Silla-Martínez & Gabaldón, 2009). To infer phylogenetic relationships we first concatenated all the alignments with FASconCAT-G v1.04 (Kück & Meusemann, 2010), constructed a maximum likelihood tree with RAxML v8.2.12, using a GAMMA model for rate heterogeneity that estimates the alpha parameter of a beta distribution of rates, and 100 bootstrap datasets to estimate statistical support of each node (Stamatakis, 2014). Secondly, we estimated a species tree via a Bayesian approach using MCMCTree implemented in PAML v4.9 (Yang, 2007). We discarded the first 2000 generations of the Markov chain as burnin and then sampled 20,000 trees, one every 20 generations, as recommended by the author. The timetree was calibrated by constraining the rabbit-guinea pig clade to a uniform temporal interval of 68 - 78 million years ago, corresponding to the TMRCA of the Hystricomorpha and the mouse-like clade (Hedges, Dudley & Kumar, 2006). The sample of posterior trees was used to generate a Hessian matrix using CODEML in PAML v4.9 and assuming a WAG+GAMMA model. The Hessian matrix was used to run MCMCTree again to obtain a Bayesian consensus tree which was visualized using the R package MCMCTreeR (Puttick, 2019). 
To predict genome-wide heterozygosity, we used Jellyfish v2.2.6 to count canonical 21-mers (Marçais & Kingsford, 2011) and GenomeScope to estimate the heterozygosity (Vurture et al., 2017). The values used for comparison were taken from Morin, et al (2020) and sources cited therein.    

**_Gene expression in the Morillo_**
To analyze genes being upregulated in the morrillo of the capybara (rostral scent marking gland), we used salmon v0.8.1 to quantify transcript abundances in each of our 11; H. hydrochaeris tissues independently against the transcriptome assembly (Patro et al, 2017). We then used edgeR to estimate dispersions and fit a negative binomial model in each of the pairwise comparisons to then fit a qCML test (Chen, Lun & Smyth, 2014). We then extracted the genes that fitted the test with a p-value < 0.001 and that had 4-fold change between the tissues, this were fitted in a matrix for which each row was a gene. We then calculated the average expression of the row and calculated relative expression patterns compared to that. This was made using Trinity scripts. The gene ontologies of the top 50 most upregulated genes in the morillo in comparison to the average expression pattern for that gene were extracted and then used as an input for REVIGO to visualize them as a scatterplot of their semantic space (Supek et al, 2011).



Citations
Ashburner, M., Ball, C. A., Blake, J. A., Botstein, D., Butler, H., Cherry, J. M., ... & Harris, M. A. (2000). Gene ontology: tool for the unification of biology. Nature genetics, 25(1), 25.
Bolger, A. M., Lohse, M., & Usadel, B. (2014). Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114-2120.
Brown, C. T., Howe, A., Zhang, Q., Pyrkosz, A. B., & Brom, T. H. (2012). A reference-free algorithm for computational normalization of shotgun sequencing data. arXiv preprint arXiv:1203.4802.
Bryant, D. M., Johnson, K., DiTommaso, T., Tickle, T., Couger, M. B., Payzin-Dogru, D., ... & Bateman, J. (2017). A tissue-mapped axolotl de novo transcriptome enables identification of limb regeneration factors. Cell reports, 18(3), 762-776.
Capella-Gutiérrez, S., Silla-Martínez, J. M., & Gabaldón, T. (2009). trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics, 25(15), 1972-1973.
Chen, Y., Lun, A. T., & Smyth, G. K. (2014). Differential expression analysis of complex RNA-seq experiments using edgeR. In Statistical analysis of next generation sequencing data (pp. 51-74). Springer, Cham.
Crusoe, M. R., Alameldin, H. F., Awad, S., Boucher, E., Caldwell, A., Cartwright, R., ... & Fenton, J. (2015). The khmer software package: enabling efficient nucleotide sequence analysis. F1000Research, 4.
Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., ... & McVean, G. (2011). The variant call format and VCFtools. Bioinformatics, 27(15), 2156-2158.
Emms, D. M., & Kelly, S. (2019). OrthoFinder: phylogenetic orthology inference for comparative genomics. BioRxiv, 466201.
El-Gebali, S., Mistry, J., Bateman, A., Eddy, S. R., Luciani, A., Potter, S. C., ... & Sonnhammer, E. L. L. (2018). The Pfam protein families database in 2019. Nucleic acids research, 47(D1), D427-D432.
Flynn, J. M., Hubley, R., Goubert, C., Rosen, J., Clark, A. G., Feschotte, C., & Smit, A. F. (2020). RepeatModeler2 for automated genomic discovery of transposable element families. Proceedings of the National Academy of Sciences, 117(17), 9451-9457.
Grabherr, M. G., Haas, B. J., Yassour, M., Levin, J. Z., Thompson, D. A., Amit, I., ... & Chen, Z. (2011). Full-length transcriptome assembly from RNA-Seq data without a reference genome. Nature biotechnology, 29(7), 644.
Guan, D., McCarthy, S. A., Wood, J., Howe, K., Wang, Y., & Durbin, R. (2020). Identifying and removing haplotypic duplication in primary genome assemblies. Bioinformatics, 36(9), 2896-2898.
Hedges, S. B., Dudley, J., & Kumar, S. (2006). TimeTree: a public knowledge-base of divergence times among organisms. Bioinformatics, 22(23), 2971-2972.
Holt, C., & Yandell, M. (2011). MAKER2: an annotation pipeline and genome-database management tool for second-generation genome projects. BMC bioinformatics, 12(1), 491.
Huerta-Cepas, J., Szklarczyk, D., Forslund, K., Cook, H., Heller, D., Walter, M. C., ... & Jensen, L. J. (2015). eggNOG 4.5: a hierarchical orthology framework with improved functional annotations for eukaryotic, prokaryotic and viral sequences. Nucleic acids research, 44(D1), D286-D293.
Jackman, S. D., Coombe, L., Chu, J., Warren, R. L., Vandervalk, B. P., Yeo, S., ... & Birol, I. (2018). Tigmint: correcting assembly errors using linked reads from large molecules. BMC bioinformatics, 19(1), 393.
Jones, P., Binns, D., Chang, H. Y., Fraser, M., Li, W., McAnulla, C., ... & Pesseat, S. (2014). InterProScan 5: genome-scale protein function classification. Bioinformatics, 30(9), 1236-1240.
Katoh, K., & Standley, D. M. (2013). MAFFT multiple sequence alignment software version 7: improvements in performance and usability. Molecular biology and evolution, 30(4), 772-780.
Kumar, S., & Subramanian, S. (2002). Mutation rates in mammalian genomes. Proceedings of the National Academy of Sciences, 99(2), 803-808.
Li, H. (2011). A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics, 27(21), 2987-2993.
Li, H., & Durbin, R. (2009). Fast and accurate short read alignment with Burrows–Wheeler transform. bioinformatics, 25(14), 1754-1760.
Li, H., & Durbin, R. (2011). Inference of human population history from individual whole-genome sequences. Nature, 475(7357), 493.
Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., ... & Durbin, R. (2009). The sequence alignment/map format and SAMtools. Bioinformatics, 25(16), 2078-2079.
Li, W., Jaroszewski, L., & Godzik, A. (2001). Clustering of highly homologous sequences to reduce the size of large protein databases. Bioinformatics, 17(3), 282-283.
Marçais, G., & Kingsford, C. (2011). A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. Bioinformatics, 27(6), 764-770.
Mistry, J., Finn, R. D., Eddy, S. R., Bateman, A., & Punta, M. (2013). Challenges in homology search: HMMER3 and convergent evolution of coiled-coil regions. Nucleic acids research, 41(12), e121-e121.
Morin, P. A., Archer, F. I., Avila, C. D., Balacco, J. R., Bukhman, Y. V., Chow, W., ... & Gulland, F. M. (2020). Reference genome and demographic history of the most endangered marine mammal, the vaquita. BioRxiv.
Patro, R., Duggal, G., Love, M. I., Irizarry, R. A., & Kingsford, C. (2017). Salmon provides fast and bias-aware quantification of transcript expression. Nature methods, 14(4), 417.
Paulino, D., Warren, R. L., Vandervalk, B. P., Raymond, A., Jackman, S. D., & Birol, I. (2015). Sealer: a scalable gap-closing application for finishing draft genomes. BMC bioinformatics, 16(1), 230.
Putnam, N. H., O'Connell, B. L., Stites, J. C., Rice, B. J., Blanchette, M., Calef, R., ... & Haussler, D. (2016). Chromosome-scale shotgun assembly using an in vitro method for long-range linkage. Genome research, 26(3), 342-350.
Puttick, M. N. (2019). MCMCtreeR: functions to prepare MCMCtree analyses and visualize posterior ages on trees. Bioinformatics, 35(24), 5321-5322.
Smit, A. F. A., Hubley, R., & Green, P. (2015). RepeatMasker Open-4.0. 2013–2015.
Stamatakis, A. (2014). RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 30(9), 1312-1313.
Supek, F., Bošnjak, M., Škunca, N., & Šmuc, T. (2011). REVIGO summarizes and visualizes long lists of gene ontology terms. PloS one, 6(7), e21800.
UniProt Consortium. (2018). UniProt: a worldwide hub of protein knowledge. Nucleic acids research, 47(D1), D506-D515.
Vurture, G. W., Sedlazeck, F. J., Nattestad, M., Underwood, C. J., Fang, H., Gurtowski, J., & Schatz, M. C. (2017). GenomeScope: fast reference-free genome profiling from short reads. Bioinformatics, 33(14), 2202-2204.
Waterhouse, R. M., Seppey, M., Simão, F. A., Manni, M., Ioannidis, P., Klioutchnikov, G., ... & Zdobnov, E. M. (2017). BUSCO applications from quality assessments to gene prediction and phylogenomics. Molecular biology and evolution, 35(3), 543-548.
Weisenfeld, N. I., Kumar, V., Shah, P., Church, D. M., & Jaffe, D. B. (2017). Direct determination of diploid genome sequences. Genome research, 27(5), 757-767. doi: 10.1101/gr.214874.116.
Weisenfeld, N. I., Yin, S., Sharpe, T., Lau, B., Hegarty, R., Holmes, L., ... & Nusbaum, C. (2014). Comprehensive variation discovery in single human genomes. Nature genetics, 46(12), 1350.
Whelan, S., Irisarri, I., & Burki, F. (2018). PREQUAL: detecting non-homologous characters in sets of unaligned homologous sequences. Bioinformatics, 34(22), 3929-3930.
Yang, Z. (2007). PAML 4: phylogenetic analysis by maximum likelihood. Molecular biology and evolution, 24(8), 1586-1591.
Yeo, S., Coombe, L., Warren, R. L., Chu, J., & Birol, I. (2017). ARCS: scaffolding genome drafts with linked reads. Bioinformatics, 34(5), 725-731.

